src/sidebar/components/test/MarkdownEditor-test.js

```diff
@@ -388,108 +385,249 @@ describe('MarkdownEditor', () => {
+
+      // Pressing arrow down, we can highlight subsequent suggestions
+      keyDownInTextarea(wrapper, 'ArrowDown');
+      assert.equal(getHighlightedSuggestion(wrapper), 1);
+      keyDownInTextarea(wrapper, 'ArrowDown');
+      assert.equal(getHighlightedSuggestion(wrapper), 2);
+
+      // Once last suggestion is highlighted, pressing arrow down has no effect
+      keyDownInTextarea(wrapper, 'ArrowDown');
+      assert.equal(getHighlightedSuggestion(wrapper), 2);
+
+      // Pressing arrow up, we can highlight preceding suggestions
+      keyDownInTextarea(wrapper, 'ArrowUp');
+      assert.equal(getHighlightedSuggestion(wrapper), 1);
+      keyDownInTextarea(wrapper, 'ArrowUp');
+      assert.equal(getHighlightedSuggestion(wrapper), 0);
+    });
+
+    it('applies highlighted suggestion when `Enter` is pressed', () => {
+      const onEditText = sinon.stub();
+      const wrapper = createComponent({
+        onEditText,
+        mentionsEnabled: true,
+        usersForSuggestions: [
+          { username: 'one', displayName: 'johndoe' },
+          { username: 'two', displayName: 'johndoe' },
+          { username: 'three', displayName: 'johndoe' },
+        ],
+      });
+
+      typeInTextarea(wrapper, '@johndoe');
+
+      // Arrow down is pressed to highlight second suggestion
+      keyDownInTextarea(wrapper, 'ArrowDown');
+      // Then Enter is pressed to apply it
+      keyDownInTextarea(wrapper, 'Enter');
+
+      // The textarea should include the username for second suggestion
+      assert.calledWith(onEditText, '@two ');
+    });
+
+    [
+      // With no users, there won't be any suggestions regardless of the text
+      { usersForSuggestions: [], text: '@', expectedSuggestions: 0 },
+
+      // With users, there won't be suggestions when none of them matches the
+      // mention
+      {
+        usersForSuggestions: [
+          { username: 'one', displayName: 'johndoe' },
+          { username: 'two', displayName: 'johndoe' },
+          { username: 'three', displayName: 'johndoe' },
+        ],
+        text: '@nothing_will_match',
+        expectedSuggestions: 0,
+      },
+
+      // With users, there will be suggestions when any of them matches the
+      // mention
+      {
+        usersForSuggestions: [
+          { username: 'one', displayName: 'johndoe' },
+          { username: 'two', displayName: 'johndoe' },
+          { username: 'three', displayName: 'johndoe' },
+        ],
+        text: '@two',
+        expectedSuggestions: 1,
+      },
+      {
+        usersForSuggestions: [
+          { username: 'one', displayName: 'johndoe' },
+          { username: 'two', displayName: 'johndoe' },
+          { username: 'three', displayName: 'johndoe' },
+        ],
+        text: '@johndoe',
+        expectedSuggestions: 3,
+      },
+    ].forEach(({ usersForSuggestions, text, expectedSuggestions }) => {
+      it('resolves expected suggestions when based on active mention matches', () => {
```

Comment:
```suggestion
      it('displays expected suggestions for mention text', () => {
```

I rewrote this to use more plain language.

Reply:
Yeah, I think the title got accidentally mangled 😅. I probably tried to rewrite it a few times and left it partially changed.

Since we have changed the "suggestion" terminology by "user", I have rewritten it to `suggests expected users for mention text`.